src/providers/ad/ad_machine_pw_renewal.c

```diff
@@ -30,10 +30,21 @@
 #define RENEWAL_PROG_PATH "/usr/sbin/adcli"
```

Comment:
Maybe rename to `RENEWAL_PROG_PATH_ADCLI` for consistency?

Reply:
It is common for FreeBSD to patch hardcoded paths like these, because we install stuff into `/usr/local` prefix by default. AFAIK, some Linux distributions install 3rd party packages into `/opt`.

Ideally the software should lookup `$PATH` or have a configure flag to allow overriding absolute paths like this one.

Reply:
Hi,

the `#define` is wrapped with `#ifndef`
````
#ifndef #ifndef RENEWAL_PROG_PATH_ADCLI
#define RENEWAL_PROG_PATH_ADCLI "/usr/sbin/adcli"
#endif

```
So if you use `CFLAGS=-DRENEWAL_PROG_PATH_ADCLI=/your/path/to/adcli` it would be already possible to override the path. But I agree that this might not be obvious and adding a configure option might be easier than documenting this method.

bye,
Sumit

Reply:
The latest version has a fourth patch with the configure options.